python/gstaichi/lang/ast/ast_transformer.py

```diff
@@ -87,6 +89,8 @@ def build_Name(ctx: ASTTransformerContext, node: ast.Name):
                     warnings.warn(message)
                 else:
                     raise exception.GsTaichiCompilationError(message)
+        if isinstance(node.ptr, Generator):
+            raise ValueError("Cannot use generator inside kernels or functions")
```

Comment:
This is misleading. It is not that generators cannot be used inside kernels or functions, but rather cannot be stored in variables, which is very different and much less problematic.